NetworkConfigurations/TrafficControlAndPolicingSubCategories/TrafficShaping.md

```diff
@@ -0,0 +1,506 @@
+
+Consider another example where a Token Bucket Filter (TBF) qdisc is attached to the second class of an Hierarchy Token Bucket (HTB) qdisc:
+
+```
+tc qdisc add dev swp24 root handle 10: htb default 1
+tc class add dev swp24 parent 10: classid 10:2 htb rate 1Gbit
+
+tc qdisc replace dev swp24 parent 10:2 handle 102: tbf rate 400Mbit burst 128K limit 1M
+```
+
+This configures an HTB qdisc on device `swp24` with a default class (`10:1`) and a second class (`10:2`) with a rate of 1Gbit. Then, a TBF qdisc is attached to the second class with a rate of 400Mbit, burst size of 128K, and a limit of 1M.
+
+```
+tc qdisc show dev swp24
+```
+
+```
+qdisc tbf 102: parent 10:2 rate 400Mbit burst 131050b lat 18.4ms
+```
+
+The output of `tc qdisc show` command displays the HTB and TBF qdiscs configured on device `swp24`, indicating their respective settings such as rates, burst sizes, and latencies.
+
+```
+tc class show dev swp24
+```
+
+```
+class htb 10:2 root leaf 102: prio 0 rate 1Gbit ceil 1Gbit burst 1375b cburst 1375b
+class tbf 102:1 parent 102:
+```
+
+The output of `tc class show` command shows the configured classes, including the HTB and TBF classes. It displays settings such as rates, burst sizes, and ceiling rates.
+
+```
+tc -s qdisc show dev swp24
+```
+
+```
+qdisc htb 10: root refcnt 2 r2q 10 default 0x1 direct_packets_stat 0 direct_qlen 1000
+ Sent 0 bytes 0 pkt (dropped 0, overlimits 0 requeues 0)
+ backlog 0b 0p requeues 0
+qdisc tbf 102: parent 10:2 rate 400Mbit burst 131050b lat 18.4ms
+ Sent 0 bytes 0 pkt (dropped 0, overlimits 0 requeues 0)
+ backlog 0b 0p requeues 0
+```
+
+The output of `tc -s qdisc show` command provides detailed information about the configured qdiscs, including statistics and settings for each. In this case, it shows the HTB and TBF qdiscs configured on the device `swp24`.
+
+Verfication using `iperf` utility:
+
+```
+iperf -c 127.0.0.1 -t 60 -b 350M
+```
+
+Please make sure to run your iperf server using `iperf -s` command.
+
+Output-
+
+```
+------------------------------------------------------------
+Client connecting to 127.0.0.1, TCP port 5001
+TCP window size: 2.50 MByte (default)
+------------------------------------------------------------
+[  3] local 127.0.0.1 port 51982 connected with 127.0.0.1 port 5001
+[ ID] Interval       Transfer     Bandwidth
+[  3]  0.0-60.0 sec  2.44 GBytes   350 Mbits/sec
+```
+
+The iperf test ran for 60 seconds, during which approximately 2.44 Gigabytes of data were transferred. The average bandwidth achieved during this test is reported as 350 Mbits/sec, which aligns with the specified bandwidth of 350 Mbps in the iperf command (-b 350M). This indicates that the traffic shaping configuration is allowing traffic to flow at the desired rate, limiting it to approximately 350 Mbps as specified.
```

Comment:
You've configured iperf to send 350 Mb/s send measured that rate. How does it prove the bandwidth is indeed limited? Send something like 400Mb/s and observe 350Mb/s.